lib/spack/spack/solver/asp.py

```diff
@@ -97,24 +99,38 @@ def getter(node):
 ast_type = ast_getter("ast_type", "type")
 ast_sym = ast_getter("symbol", "term")
 
-#: Order of precedence for version origins. Topmost types are preferred.
-version_origin_fields = [
-    "spec",
-    "dev_spec",
-    "external",
-    "packages_yaml",
-    "package_requirements",
-    "package_py",
-    "installed",
-]
 
-#: Look up version precedence strings by enum id
-version_origin_str = {i: name for i, name in enumerate(version_origin_fields)}
+class Provenance(enum.IntEnum):
```

Comment:
> I assume you specifically wanted to use IntEnum because these values are weights. 

They are not weights directly, but they are used to determine weights. Specifically, in:

https://github.com/spack/spack/blob/5bb5308e4b90fcf7fbd7034b7fd6a060d76cb560/lib/spack/spack/solver/asp.py#L906-L913

we sort versions by provenance first. A "regular" enum doesn't define the `__lt__` operator that is needed for sorting, while an int enum does it for free.